mysql-test/lib/mtr_report.pm

```diff
@@ -499,31 +501,49 @@ sub mtr_report_stats ($$$$) {
 
       my $test_result;
 
-      # if a test case has to be retried it should have the result MTR_RES_FAILED in jUnit XML
       if ($test->{'retries'} > 0) {
-        $test_result = "MTR_RES_FAILED";
+        $test_result = MTR_RES_FAILED;
+        if ($test->{'failures'} > 1){
+          $test_result = MTR_RES_RETRY_FAILED;
+        }
+        if ($test->{'result'} eq MTR_RES_PASSED){
+          $test_result = MTR_RES_RETRY_PASSED;
+        }
       } else {
         $test_result = $test->{'result'};
       }
+      if ($test_result eq MTR_RES_PASSED && defined($test->{'warnings'})){
+        $test_result = MTR_RES_PASSED_WARNINGS;
+      }
+
+      my $combinations;
+      if (defined($test->{combinations})){
+        $combinations = join ',', @{$test->{combinations}};
+      } else {
+        $combinations = "";
+      }
 
-      $xml_report .= qq(\t\t<testcase assertions="" classname="$current_suite" name="$test->{'name'}" status="$test_result" time="$test_time");
+      $xml_report .= qq(\t\t<testcase assertions="" classname="$current_suite" name="$test->{'name'}" ).
+                     qq(status="$test_result" time="$test_time" combinations="$combinations");
 
-      my $comment = $test->{'comment'};
-      $comment =~ s/[\"]//g;
+      my $warnings= replace_special_symbols($test->{'warnings'});
+      my $comment= replace_special_symbols($test->{'comment'});
 
-      # if a test case has to be retried it should have the result MTR_RES_FAILED in jUnit XML
-      if ($test->{'result'} eq "MTR_RES_FAILED" || $test->{'retries'} > 0) {
+      if ($test->{'result'} eq MTR_RES_FAILED) {
         my $logcontents = $test->{'logfile-failed'} || $test->{'logfile'};
+        $logcontents= $logcontents.$test->{'warnings'}."\n";
```

Comment:
My opinion is that the only line needed by description of MDEV is this one. 
Passing to Serg for further review.

Reply:
should other changes be moved to another PR?

Reply:
I would say no need, if there is an agreement to have new options in XML format. In that case I think correction of description in MDEV in JIRA would be enough.

Reply:
MDEV is now "Valgrind / ASAN warnings and retry-pass/retry-fail info do not go to MTR XML, XML can contain illegal symbols"